src/presentational-components/shared/table-composable-toolbar-view.tsx

```diff
@@ -0,0 +1,260 @@
+          sortBy={orderBy}
+          pagination={pagination}
+          fetchData={fetchData}
+          toolbarButtons={toolbarButtons}
+          filterPlaceholder={filterPlaceholder}
+          filters={filters}
+          isFilterable={isFilterable}
+          onShowMore={onShowMore}
+          showMoreTitle={showMoreTitle}
+          onFilter={onFilter}
+          onChange={onChange}
+          value={value}
+          hideFilterChips={hideFilterChips}
+          tableId={tableId}
+          textFilterRef={textFilterRef}
+        />
+        {isLoading ? (
+          <ListLoader isCompact={isCompact} items={pagination?.limit} />
+        ) : (
+          <TableComposable
+            aria-label={`${title.plural.toLowerCase()} table`}
+            variant={isCompact ? TableVariant.compact : undefined}
+            borders={borders}
+            ouiaId={ouiaId} // [PF]: Value to overwrite the randomly generated data-ouia-component-id
+          >
+            <Thead>
+              <Tr>
+                {columns.map((column, i) => (
+                  <Th key={i} sort={column?.sortable ? { columnIndex: i, sortBy, onSort } : undefined}>
+                    {column.title}
+                  </Th>
+                ))}
+              </Tr>
+            </Thead>
+            <Tbody>
+              {rows?.length > 0 ? (
+                rows?.map((row, i) => (
+                  <Tr key={i}>
+                    {row.cells.map((cell, j) => (
+                      <Td key={j} dataLabel={columns[j].title}>
+                        {/* TODO: make more general */}
+                        {isCellObject(cell) ? cell.title : cell}
+                      </Td>
+                    ))}
+                  </Tr>
+                ))
+              ) : (
+                <Tr>
+                  {/* render one component full width for the entire row */}
+                  <Td colSpan={columns.length}>{renderEmpty()}</Td>
+                </Tr>
+              )}
+            </Tbody>
+          </TableComposable>
+        )}
+        {!pagination.noBottom && (
+          <TableToolbar>
+            {!isLoading && <Pagination {...paginationBuilder(pagination, fetchData, filterValue, orderBy)} variant="bottom" dropDirection="up" />}
+          </TableToolbar>
+        )}
+      </Fragment>
+    );
+  };
+
+  return (
+    <Fragment>
+      {!isLoading && rows?.length === 0 && filterValue?.length === 0 && filters.every(({ value }) => !value) ? (
+        <EmptyWithAction
+          title={intl.formatMessage(messages.configureItems, { items: title.plural })}
+          icon={PlusCircleIcon}
+          description={[
+            intl.formatMessage(messages.toConfigureUserAccess),
+            intl.formatMessage(messages.createAtLeastOneItem, { item: title.singular }),
+          ]}
+          actions={toolbarButtons ? toolbarButtons()[0] : false}
+          {...(typeof emptyProps === 'object' ? emptyProps : {})}
+        />
+      ) : (
+        renderTable()
```

Comment:
In general we should move away from these render functions. It can cause trouble in React 18. We have two options:

1. Move the entire JSX and remove the wrapping render functions.
2. Create new component and pass all relevant props to it.

Reply:
That is a good point! thanks